hdf/test/nbit.c

```diff
@@ -571,54 +558,51 @@ test_nbit6(int32 fid)
 static void
 test_nbit7(int32 fid)
 {
-    int32      aid1;
-    uint16     ref1;
-    int        i;
-    int32      ret;
-    intn       errors = 0;
-    model_info m_info;
-    comp_info  c_info;
-    uint8     *outbuf, *inbuf;
```

Comment:
I feel like it would be a lot easier to initialize inbuf and outbuf to NULL, then free them unconditionally at the end.

Also, I'd like for the code to look like HDF5 to minimize friction moving between the libraries. In HDF5, all declarations are at the top of the function, things that need to be freed are initialized to "bad" values, and we free things at the end when they are non-bad (if the cleanup code can't accept bad values, otherwise unconditionally).

I know there are arguments to the distributed declaration style, but "everything at the top" is what we decided on in HDF5 a long time ago. It's also a LOT easier to ensure that everything allocated in the function is disposed of at the end. In this style of code, you have to read every line of code to do the same thing and cleaning up multiple allocated resources can get really messy.